### ISSUE TITLE
Simplify (x^1) != 0

### ISSUE THREAD

**Issue by joelreymont** [Vector35/binaryninja-api#5545]
**Version and Platform (required):**
- Binary Ninja Version: 4.1.5428-dev, 5597859f
- OS: macos
- OS Version: 14.5
- CPU Architecture: arm64

Internal binary `major dine favor`.

In `DjiAircraftInfo_GetAircraftTypeByCameraVersion`...

I believe this 

```
00047d64          if (zx.d(DjiError_IsSuccess(err_code) ^ 1) != 0)
```

should be written `!= 1` for clarity. 

IDA does this 

```
    if ( !DjiError_IsSuccess_localalias_0(err_code) )
```

**Comment by xusheng6** [comment]
Related to https://github.com/Vector35/binaryninja-api/issues/5526 and https://github.com/Vector35/binaryninja-api/issues/5525

**Comment by negasora** [comment]
Fixed in `4.2.5894-dev`
